### PR TITLE
JdsIcon: Create component and refactor other components' icon usage

### PR DESCRIPTION
**Pull request by adrianpdm**
### Changes
- Add all icon from Figma as SVG assets (action, caret, chevron, object, socmed) (70 items)
- Add `svg-to-vue` loader, so SVG can be imported as Vue component
- Create JdsIcon component
- Create JdsIcon stories
- Refactor other components icon usage (InputTextEdge, CheckboxToggle)

### Note
- JdsIcon has no dedicated event yet

### Preview
image

### CODE REVIEW

File: src/components/JdsIcon/Icon.vue [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<template>
+  <i
+    :class="iconClasses"
+    v-on="$listeners"
```
Comment (adzharamrullah) [inline src/components/JdsIcon/Icon.vue:4]: found a litte bit confusing where variable `$listeners` came from ?
i can't see that variable define on below, do you explain about that ?

Comment (adrianpdm) [reply src/components/JdsIcon/Icon.vue:4]: `$listeners` can be found [here](https://vuejs.org/v2/api/#vm-listeners).

`$listeners` can be used on root element to make a transparent wrapper for event listeners.
All event listeners define on its parent will automatically be assigned to the component,
without having to manually call $emit.

In parent/consumer,
```html
<template>
   <jds-icon @click="onClick" @focus="onFocus" @mouseover="onMouseOver" @keydown="onKeydown" />
</template>
``` 

In JdsIcon
```html
<template>
   <i class="iconClasses" v-on="$listeners">
       <!-- this root element will listen to click, focus, mouseover, keydown, as defined in its parent -->
   </i>
</template>
```

Comment (adrianpdm) [reply src/components/JdsIcon/Icon.vue:4]: without using v-on="$listeners",
we usually have to use `native` modifier on its parent/consumer.

In JdsIcon
```html
<template>
   <i class="iconClasses" >
       <!-- this root element is not using $listeners -->
   </i>
</template>
```

In parent/consumer,
```html
<template>
   <jds-icon
     @click.native="onClick"
     @focus.native="onFocus"
     @mouseover.native="onMouseOver"
     @keydown.native="onKeydown"
   />
</template>
```

**Review by adzharamrullah — COMMENTED** [review]
After i saw all of icon, i think some icon like 
- magnifier
- open-new-tab
- cloud-download
- door-exit

can give a suffix `outline`, even though there is no icon solid for it icon, maybe later can be already

File: src/components/JdsIcon/Icon.vue (hunk 1 above)
Comment (adzharamrullah) [reply src/components/JdsIcon/Icon.vue:4]: That was great, cool 😱 i just found out about it

**Comment by adrianpdm** [comment]
> After i saw all of icon, i think some icon like
> 
> * magnifier
> * open-new-tab
> * cloud-download
> * door-exit
> 
> can give a suffix `outline`, even though there is no icon solid for that icon, maybe later can be already

I was a bit confused on whether an icon classified as an `outline icon` or a `solid icon`.

I think icon naming shouldn't have been my duty lol :running_man:

**Comment by adzharamrullah** [comment]
> > After i saw all of icon, i think some icon like
> > 
> > * magnifier
> > * open-new-tab
> > * cloud-download
> > * door-exit
> > 
> > can give a suffix `outline`, even though there is no icon solid for that icon, maybe later can be already
> 
> I was a bit confused on whether an icon classified as an `outline icon` or a `solid icon`.
> 
> I think icon naming shouldn't have been my duty lol 🏃‍♂️

If you think icon naming it's not your duty, i thinks i can resolve this conversation, lol 😄

**Review by adzharamrullah — APPROVED** [review]

**Review by Ibwedagama — APPROVED** [review]

**Review by yoslie — APPROVED** [review]

**Review by Arif9878 — APPROVED** [review]

**Pull request merged**